react-todo-app/src/TodoContainer/Components/TodoListItem/TodoListItem.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useRef, useEffect } from "react";
+import style from "./TodoListItem.module.css";
+import PropTypes from "prop-types";
+import { Tooltip, List, Checkbox, Input, Typography } from "antd";
+import { DeleteOutlined, FormOutlined } from "@ant-design/icons";
+const { TextArea } = Input;
+
+// import { requestEditCheck } from "../../API";
+
+const TodoListItem = ({
+  todoList,
+  onRemoveTodo,
+  onEditTodo,
+  handleDescription,
+  tableName,
+}) => {
+  const [isToggle, setToggle] = useState(false);
+  const [todoEditTitle, setTodoEditTitle] = useState(
+    todoList.fields.Title || ""
+  );
+  const editInputRef = useRef(null);
+  const listItemRef = useRef(null);
+
+  const onChangeEdit = (e) => {
+    const editTodo = e.target.value;
+    setTodoEditTitle(editTodo);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    if (todoEditTitle.trim().length === 0) {
+      alert("Action cannot be blank");
+    } else {
+      onEditTodo(
+        todoList.id,
+        {
+          fields: {
+            Title: todoEditTitle,
+            Description: todoList.fields.Description,
+            Done: todoList.fields.Done,
+          },
+        },
+        tableName
+      );
+      setToggle(false);
+    }
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (e) => {
+      if (
+        listItemRef.current &&
+        !listItemRef.current.contains(e.target) &&
+        editInputRef.current !== e.target
+      ) {
+        setToggle(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  const [isChecked, setIsChecked] = useState(todoList.fields.Done);
+
+  const handleCheckBoxChange = async () => {
```

Comment:
I'm expecting that the todo item will be striked through immediately when I click a checkbox. Is that the intended behavior?

I'm seeing the todo field go into an editing state when I click the checkbox
![image](https://user-images.githubusercontent.com/6400323/233339539-add3ba69-e011-4e59-9ab6-f3f6be37e425.png)